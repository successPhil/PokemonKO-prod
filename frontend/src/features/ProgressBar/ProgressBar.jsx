import React from 'react';
import { motion } from 'framer-motion';
import GameText from '../../styles/PokemonGB';

const ProgressBar = ({ value, maxValue, barColor }) => {
  const percentage = (value / maxValue) * 100;

  return (
    <div
      style={{
        width: '500px',
        height: '31px',
        background: 'rgba(255, 255, 255, 0.5)',
        borderRadius: '16px',
        overflow: 'hidden',
        position: 'relative',
      }}
    >
      <motion.div
        style={{
          width: '100%',
          height: '100%',
          background: barColor || '#4caf50',
          borderRadius: '16px',
        }}
        initial={{ width: 0 }}
        animate={{ width: `${percentage}%` }}
        transition={{ ease: "easeInOut", duration: 2 }}
      >
        <GameText
        className='bar'
          style={{
            position: 'absolute',
            left: '50%',
            top: '50%',
            transform: 'translate(-50%, -50%)',
            color: '000000',
            // visibility: 'hidden', // Initially hide the text
          }}
        >
          {`${value}/${maxValue}`}
        </GameText>
      </motion.div>
    </div>
  );
};

export default ProgressBar;
