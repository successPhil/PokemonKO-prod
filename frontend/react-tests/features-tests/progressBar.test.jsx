import ProgressBar from "../../src/features/ProgressBar/ProgressBar";
import { expect, test } from 'vitest'

test("Progress Bar renders without crashing", ()=>{
    const progressBar = <ProgressBar />
    expect(progressBar).toMatchSnapshot()
})