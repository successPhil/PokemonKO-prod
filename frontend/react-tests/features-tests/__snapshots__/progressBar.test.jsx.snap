// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Progress Bar renders without crashing 1`] = `<ProgressBar />`;
